src/routes/SafeAppLandingPage/SafeAppLandingPage.tsx

```diff
@@ -75,6 +75,10 @@ const SafeAppLandingPage = (): ReactElement => {
 
   const showLoader = isLoading || !safeAppDetails
 
+  if (!safeAppUrl) {
+    return <Redirect to={WELCOME_ROUTE} />
```

Comment:
This edge case is already addressed [in the first useEffect](https://github.com/gnosis/safe-react/pull/3790/files#diff-5c58fd1e35b170b805ce84c3adfb087f547f39ae5a3cac1f2a715ce6785888c4R29) right?

What is your opinion about to remove this redundant redirection and keep showing the Loader? (and restore [this 2 lines](https://github.com/gnosis/safe-react/pull/3790/files#diff-af57d018588c827d3361cb533face28224d494af4621c96c30e3b910c8bec019L162) in the unit tests?)

Reply:
I don't think addressing it in `useEffect` is the way to go. We don't need to render anything if there's no `safeAppUrl`. This is also better for types stuff, because `query.get` returns `string | undefined` and this narrows it to `string`, making it easer for subsequent access

Reply:
Then maybe makes sense update both `useEffect` ?. Something like this:

```
const SafeAppLandingPage = (): ReactElement => {
  const query = useQuery()
  const safeAppChainId = query.get('chainId')
  const safeAppUrl = query.get('appUrl')

  useEffect(() => {
    if (isValidChainId(safeAppChainId)) {
      setChainId(safeAppChainId)
    }
  }, [safeAppChainId])

  [...] // other stuff

  // the second useEffect is unnecessary, and instead, do:

  const isSafeAppMissing = !isLoading && !safeAppDetails && isManifestError

  if (!safeAppUrl || !safeAppChainId || isSafeAppMissing) {
    return <Redirect to={WELCOME_ROUTE} />
  }

  const availableChains = safeAppDetails?.chainIds || []

  const showLoader = isLoading || !safeAppDetails

  return (
    <Container>
      <StyledCard>
            [...]
```

Reply:
done senor